### PR TITLE
[15.0][MIG] crm_stage_probability

### PR DESCRIPTION
**Pull request by vib-adhoc**

### CODE REVIEW

**Review by jjscarafia — APPROVED** [review]
:+1: code review

File: crm_stage_probability/tests/test_crm_lead_probability.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        cls.stage_qualified = cls.env.ref("crm.stage_lead2")
+        cls.stage_proposition = cls.env.ref("crm.stage_lead3")
+        cls.stage_won = cls.env.ref("crm.stage_lead4")
+        cls.opportunity_1 = cls.env.ref("crm.crm_case_32")
+
+    def test_update_probability(self):
+        self.assertEqual(self.opportunity_1.stage_id, self.stage_qualified)
+        self.assertFalse(self.opportunity_1.is_automated_probability)
+        self.assertFalse(self.opportunity_1.is_stage_probability)
+        self.opportunity_1.write({"stage_id": self.stage_new.id})
```
Comment (pere-aquarian) [inline crm_stage_probability/tests/test_crm_lead_probability.py:21]: Hi, @vib-adhoc !
I have reviewed the codecov messages and maybe an oportunity with a null stage_id could make test entering in some missing lines.

**Comment by vib-adhoc** [comment]
Hi @pere-aquarian ! I added some tests acording your suggestion.

File: crm_stage_probability/tests/test_crm_lead_probability.py [hunk 2]
```diff
@@ -47,6 +52,13 @@ def test_create_opportunity(self):
...
         self.assertFalse(opportunity.is_automated_probability)
+        # test if there is no default stage
+        self.env["crm.stage"].search([("fold", "=", False)]).write({"fold": True})
+        opportunity = self.env["crm.lead"].create(
+            {"name": "My opportunity", "type": "opportunity"}
+        )
+        self.assertEqual(opportunity.probability, 10)
+        self.assertFalse(opportunity.is_automated_probability)
 
     def test_create_opportunity_default_stage_id(self):
```
Comment (pere-aquarian) [inline crm_stage_probability/tests/test_crm_lead_probability.py:66]: Hi, @vib-adhoc !

Seem that the test needs to include a call to onchange_stage_id_values . So maybe a new method that invoques that could solve the problem. 
Maybe a def test_change_stage_id_values(self) with a simple event that calls onchange_stage_id_values

Thank you!

Comment (vib-adhoc) [reply crm_stage_probability/tests/test_crm_lead_probability.py:66]: Hi @pere-aquarian ! I added a test, thanks for your suggestion.

Comment (pere-aquarian) [reply crm_stage_probability/tests/test_crm_lead_probability.py:66]: thanks a lot, @vib-adhoc and @jjscarafia

**Review by pere-aquarian — APPROVED** [review]
LGTM!

**Pull request merged**